src/prime_server.cpp

```diff
@@ -263,18 +267,28 @@ bool server_t<request_container_t, request_info_t>::enqueue(const zmq::message_t
   for (const auto& parsed_request : parsed_requests) {
     // figure out if we are expecting to close this request or not
     auto info = parsed_request.to_info(request_id++);
-    // send on the request
-    if (!proxy.send(static_cast<const void*>(&info), sizeof(info), ZMQ_DONTWAIT | ZMQ_SNDMORE) ||
-        !proxy.send(parsed_request.to_string(), ZMQ_DONTWAIT)) {
+
+    // if its enabled, see if its a health check
+    bool health_check = health_check_matcher && health_check_matcher(parsed_request);
+
+    // send on the request if its not a health check
+    if (!health_check &&
+        (!proxy.send(static_cast<const void*>(&info), sizeof(info), ZMQ_DONTWAIT | ZMQ_SNDMORE) ||
+         !proxy.send(parsed_request.to_string(), ZMQ_DONTWAIT))) {
       logging::ERROR("Server failed to enqueue request");
       return false;
     }
     if (log)
       parsed_request.log(info.id);
+
     // remember we are working on it
     request.enqueued.emplace_back(static_cast<typename decltype(requests)::key_type>(info));
     requests.emplace(request.enqueued.back(), requester);
     request_history.emplace_back(std::move(info));
+
+    // if it was a health check we reply immediately
+    if (health_check)
+      dequeue(request_history.back(), health_check_response);
```

Comment:
this is the actual change to enable the functionality. i didnt go crazy here. if you opted into the health check functionality and we see a request is a health check, we dont forward it to the proxy/load balancer, instead we enqueue as if we did but then when immediately dequeue it (which is what replies to the client) with the canned response. i could have avoided enquing and dequing it and just replied directly but liked that the logging and also the connection closing logic is encapsulated in those functions so i kept it simple at the sake of a tiny bit of performance degradation